src/app/CommandHandler.h

```diff
@@ -48,10 +52,32 @@ class CommandHandler : public Command
                              const Protocols::SecureChannel::GeneralStatusCode aGeneralCode, const Protocols::Id aProtocolId,
                              const Protocols::InteractionModel::Status aStatus) override;
 
+    /**
+     * API for adding a data response.  The template parameter T is generally
+     * expected to be a ClusterName::Commands::CommandName::Type struct, but any
+     * struct that can be encoded using our Encode machinery and exposes the
+     * right command id will work.
+     *
+     * @param [in] aRespondingTo the concrete path of the command we are
+     *             responding to.
+     * @param [in] aFields the data fields for the response.
+     */
+    template <typename FieldStruct>
+    CHIP_ERROR AddDataResponse(const ConcreteCommandPath & aRespondingTo, const FieldStruct & aFields)
```

Comment:
AddDataResponse implies that after this call, the response is sent.

A slight re-order gives a better implication of just the encoding of the command data, which aligns with the way status code is worded as well.

```suggestion
    CHIP_ERROR AddResponseData(const ConcreteCommandPath & aRespondingTo, const FieldStruct & aFields)
```

Reply:
Done.